core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -731,9 +731,10 @@ class KafkaApis(val requestChannel: RequestChannel,
         // Prepare fetch response from converted data
         val response = new FetchResponse(unconvertedFetchResponse.error(), convertedData, throttleTimeMs,
           unconvertedFetchResponse.sessionId())
-        response.responseData.asScala.foreach { case (topicPartition, data) =>
-          // record the bytes out metrics only when the response is being sent
-          brokerTopicStats.updateBytesOut(topicPartition.topic, fetchRequest.isFromFollower, data.records.sizeInBytes)
+        // record the bytes out metrics only when the response is being sent
+        response.responseData.asScala.foreach { case (tp, data) =>
+          brokerTopicStats.updateBytesOut(tp.topic, fetchRequest.isFromFollower,
+            replicaManager.isAddingReplica(tp, fetchRequest.replicaId()), data.records.sizeInBytes)
```

Comment:
In a separate iteration, I thought we had added `isAddingReplica` to `FetchPartitionData`? It's probably not _that_ big of a deal, but it would be nice not to need to dip back into `ReplicaManager` on every fetch. If it doesn't work out, we could at least skip this for consumer fetches.

Reply:
The main problem here is `FetchPartitionData` represents the state before any conversion occurs, therefore if we count the metrics there then it won't account for and record size difference created by downconversion.
Maybe it works if I create a `FetchResponse.PartitionData -> Boolean` map in `processResponseCallback` which acts as the intermediary for this purpose but for first it seems a bit complicated. I'll upload a commit with these modifications.